src/main/java/com/name/OX/GameMain.java

```diff
@@ -15,7 +15,7 @@
 
     public static void main(String[] args) {
         GameMain gameMain = new GameMain(new OptionsState(new GameOptions(), new Scanner(System.in)));
-        while (true){
+        while (!currentState.whichState().equals(GameStateName.FINISH)){
```

Comment:
Jak zmieniliśmy ten while condition to chyba ten nullek + exit(0) co zaznaczyłem powyżej już nie jest potrzebny? ;)